device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsSessionDeviceOperation.java

```diff
@@ -337,6 +337,13 @@ boolean onLinkRemoteOpen(String linkName)
             {
                 if (entry.getValue().onLinkRemoteOpen(linkName))
                 {
+                    if (entry.getKey() == MessageType.DEVICE_TWIN)
```

Comment:
actually, you do need a check here that the opened link was a sender link. If you try to send this message when only the receiver link is open, an exception will get thrown. Plus you don't want to send this subscribe message once when the sender link opens, and again once the receiver link opens